static/js/apps/nl_interface/app.tsx

```diff
@@ -44,8 +45,29 @@ export function App(): JSX.Element {
     return () => clearTimeout(timer);
   }, [queries]);
 
+  function addContext(context: any, idx: number) {
+    // Always assume we are appending context for the latest query.
+    if (idx !== queries.length - 1) {
+      console.error(
+        "setting context for wrong length: idx:%i != queries:%i",
+        idx,
+        queries.length
+      );
+      return;
+    }
+    const newList = [...contextHistory];
+    newList.push(context);
+    setContextList(newList);
+  }
+
   const queryResults = queries.map((q, i) => (
-    <QueryResult key={i} query={q}></QueryResult>
+    <QueryResult
```

Comment:
from testing, do you see the error case. Or if the component here can always invoke the callback in order?

Reply:
i haven't seen the error case. but wanted to prevent it from getting wildly out of sync..

Reply:
i take it back. it happens when you change the embeddings build. in which case we shouldn't update the context?

Reply:
I think this works for now. We can revisit if errors do occur.